src/design-system/components/chip/Chip.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { CSSProperties, useState } from 'react'
+import classNames from 'classnames'
+import CircleIcon from '@mui/icons-material/Circle'
+import CloseRoundedIcon from '@mui/icons-material/CloseRounded'
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
+
+import './Chip.scss'
+
+type ChipContentProps = {
+  label: string
+  startIcon?: React.ReactNode
+  endIcon?: React.ReactNode
+}
+
+type ChipType = 'filter' | 'input' | 'category' | 'state'
+
+type BaseChipProps = ChipContentProps & {
+  type: ChipType
+  size?: 'sm' | 'md' | 'lg'
+  disabled?: boolean
+}
+
+type FilterChipProps = BaseChipProps & {
+  type: 'filter'
+  options: string[]
+  variant?: 'filled' | 'minimal'
+}
+
+type InputChipProps = BaseChipProps & {
+  type: 'input'
+  variant?: 'filled' | 'outlined' | 'filled-outlined'
+}
+
+type ChipColor =
+  | 'blue-gray'
+  | 'turquoise'
+  | 'blue'
+  | 'green'
+  | 'yellow'
+  | 'purple'
+  | 'red'
+  | 'neutral'
+
+type CategoryChipProps = BaseChipProps & {
+  type: 'category'
+  detail: string
+  color?: ChipColor
+}
+
+type StateChipProps = BaseChipProps & {
+  type: 'state'
+  color?: ChipColor
+}
+
+export type ChipProps =
+  | FilterChipProps
+  | InputChipProps
+  | CategoryChipProps
+  | StateChipProps
+
+type CustomStyle = CSSProperties & {
+  [key: `--chip-${string}`]: string | undefined
+}
+
+const ChipContent: React.FC<ChipContentProps> = ({
+  label,
+  startIcon,
+  endIcon
+}) => (
+  <>
+    {startIcon && <span className='startIcon'>{startIcon}</span>}
+    <span className='label'>{label}</span>
+    {endIcon && <span className='endIcon'>{endIcon}</span>}
+  </>
+)
+
+const FilterChip: React.FC<FilterChipProps> = ({
+  label,
+  options = [],
+  variant = 'filled',
+  startIcon = <CircleIcon style={{ fontSize: 'inherit' }} />,
+  endIcon = <ExpandMoreIcon style={{ fontSize: 'inherit' }} />,
+  disabled = false,
+  size = 'md'
+}) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const [selectedOption, setSelectedOption] = useState<string | null>(null)
+
+  const handleSelect = (option: string) => {
+    setSelectedOption(option)
+    setIsOpen(false)
+  }
+
+  const isSelected = !!selectedOption
```

Comment:
```suggestion
  const isSelected = Boolean(selectedOption)
```
It is more clean, I think

Reply:
agree